packages/web/hooks/input/use-search-query-input.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useMemo, useState } from "react";
+
+import type { Search } from "~/utils/search";
+
+import { useDebouncedState } from "../use-debounced-state";
+
+/** Manages input for a search box that will be used
+ *  to perform a search remotely (is debounced). */
+export function useSearchQueryInput(debounceMs = 500) {
+  // get selectable currencies for trading, including user balances if wallect connected
+  const [searchInput, setSearchInput] = useState<string>("");
+
+  // generate debounced search from user inputs
+  const [debouncedSearchInput, setDebouncedSearchInput] =
+    useDebouncedState<string>("", debounceMs);
+  useEffect(() => {
+    const sanitizedSearchInput = searchInput.replace(/#/g, "").trim();
+    setDebouncedSearchInput(sanitizedSearchInput);
+  }, [setDebouncedSearchInput, searchInput]);
```

Comment:
Is there any reason why we are using an effect instead of creating a callback function wrapping setDebouncedSearchInput?

Reply:
Thanks for noticing this😅, seems I got carried away with some of the effects in this one